src/main/java/org/jabref/model/database/BibDatabaseContext.java

```diff
@@ -198,7 +198,13 @@ public boolean isBiblatexMode() {
 
         // 4. BIB file directory
         getDatabasePath().ifPresent(dbPath -> {
-            String parentDir = dbPath.getParent().toAbsolutePath().toString();
+            assert dbPath != null : "dbPath is null";
```

Comment:
Sure it is :). IMHO, assert seems a bit more applicable here, since I see asserts intended for developers, and Objects.requireNonNull() will probably end up with the message for the end user.
The assert statements are off by default but can be switched on 'in the field', unless the they were deliberately optimised away. As I understand, Objects.requireNonNull() will be always n and will always run. If this is what your policy is, I can change it, for sure.